packages/@react-spectrum/toast/src/ToastContainer.tsx

```diff
@@ -166,7 +169,27 @@ function addToast(children: string, variant: SpectrumToastValue['variant'], opti
   // It is debatable whether non-actionable toasts would also fail.
   let timeout = options.timeout && !options.onAction ? Math.max(options.timeout, 5000) : undefined;
   let queue = getGlobalToastQueue();
-  let key = queue.add(value, {timeout, onClose: options.onClose});
+  let key: string;
+  let add = () => queue.add(value, {timeout, onClose: options.onClose});
+  if ('startViewTransition' in document) {
+    // @ts-ignore
+    document.startViewTransition(() => {
+      flushSync(() => {
+        key = add();
+      });
+    });
+  } else {
+    key = add();
+  }
+
+  if ('startViewTransition' in document) {
```

Comment:
I'm not sure why there are two startViewTransitions, could you add some code comments explaining what is happening here?

also, any reason not to move this up into the other if above?

Reply:
one of them is adding the toast and the other is removing it I think

Reply:
Yep, since we return the `close` function, we wrap that too